Library/Homebrew/dev-cmd/extract.rb

```diff
@@ -133,7 +133,7 @@ def run
         result.sub!("class #{class_name} < Formula", "class #{versioned_name} < Formula")
 
         # Remove bottle blocks, as they won't work.
```

Comment:
Can the bottles be used? If not, can you please elaborate: what's the value in noting where one can find them?

Also, `brew extract` can be used on formulae from taps other than `homebrew/core`, so this may not be accurate.

Reply:
> Can the bottles be used?

As in re-used from the original formula? No, because after the extraction the name of the formula gets changed from `<formula>` to `<formula>@<version>` and thus the URL gets changed from `docker://ghcr.io/homebrew/core/<name>:<version>` to `docker://ghcr.io/homebrew/core/<name>/version:<version>`.

> what's the value in noting where one can find them

When the intent is to pin formula for a reproducible build, it's important to archive the bottles as well, so one could use Homebrew (an old version, if necessary) with a custom tap and a custom `HOMEBREW_BOTTLE_DOMAIN`.

As it stands it's hard / impossible to entirely automate the process, e.g. via imaginary `brew extract --include-bottles` because:

1. Bottles may not be present
2. Formula may come from a custom tap

This hint is the next best thing to do as it puts the developer right on track of getting it to work. I'd greatly appreciate it as user.

Perhaps a better solution would be to either document the entire workflow of "taking a snapshot" of a given formula or, at least, document how Homebrew stores and retrieves the bottles.

Reply:
Could you just rename the formula to not include the `@version` part?

> As it stands it's hard / impossible to entirely automate the process, e.g. via imaginary `brew extract --include-bottles` because:
> 
> 1. Bottles may not be present
> 2. Formula may come from a custom tap

We can simply restrict `--include-bottles` to homebrew/core formulae, and fail if there are no bottles (or, maybe, still extract the formula and just error on the bottle part). I used to have a custom brew command that just downloaded and extracted bottles [here](https://github.com/Rylan12/homebrew-unpack-bottle). I stopped using it and I don't think it works anymore, but it could maybe serve as a starting point to implementing something like this.

Reply:
@Rylan12 Still there is a problem that Homebrew, being rolling-release package manager, may decide to abandon old bottles. 

Instead of `--include-bottles` I'd rather prever `--extract-bottles` so I could store them elsewhere and not depend on benevolence of the original formula maintainer.

Reply:
*please also see this discussion: https://github.com/orgs/Homebrew/discussions/5454